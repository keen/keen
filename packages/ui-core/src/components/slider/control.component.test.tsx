import React from 'react';
import { shallow } from 'enzyme';
import Control from './control.component';

describe('<Control />', () => {
  const props = {
    initialPos: 0,
    sliderSize: 200,
    min: 0,
    max: 100,
    dragConstraints: {
      top: 0,
      left: 0,
      right: 100,
      bottom: 100,
    },
    steps: 1,
    tooltip: {
      enabled: true,
      position: 'top',
    },
  };

  test('should render Control component', () => {
    const wrapper = shallow(<Control isHorizontal={true} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
