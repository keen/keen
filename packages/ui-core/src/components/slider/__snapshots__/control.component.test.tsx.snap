// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Control /> should render Control component 1`] = `
<Fragment>
  <ForwardRef(MotionComponent)
    drag="x"
    dragConstraints={
      Object {
        "bottom": 100,
        "left": 0,
        "right": 100,
        "top": 0,
      }
    }
    dragElastic={0}
    dragMomentum={false}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": "2px solid #CA8917",
        "borderRadius": "50%",
        "boxSizing": "border-box",
        "height": 16,
        "left": -7,
        "position": "absolute",
        "top": NaN,
        "width": 16,
        "x": MotionValue {
          "canTrackVelocity": true,
          "current": 0,
          "lastUpdated": 0,
          "notifySubscriber": [Function],
          "parent": undefined,
          "prev": undefined,
          "scheduleVelocityCheck": [Function],
          "setChild": [Function],
          "timeDelta": 0,
          "transformer": undefined,
          "updateAndNotify": [Function],
          "velocityCheck": [Function],
        },
        "zIndex": undefined,
      }
    }
  />
</Fragment>
`;
